neurox/data/writer.py

```diff
@@ -166,11 +178,17 @@ def write_activations(self, sentence_idx, extracted_words, activations):
         if self.decompose_layers:
             for writer_idx, layer_idx in enumerate(self.layers):
                 self.writers[writer_idx].write_activations(
-                    sentence_idx, extracted_words, activations[layer_idx, :, :]
+                    sentence_idx,
+                    extracted_words,
+                    activations[layer_idx, :, :],
+                    dtype=self.dtype,
```

Comment:
Can you add `dtype` as a member of the base `ActivationsWriter` class? this way the `write_activations` signature will not need to be changed (`write_activations` can use `self.dtype` instead of using an argument)

Conceptually, arguments of `write_activations` must be things that may change with every write, and not things that are constant for the current session like dtype

Reply:
Done. dtype is still be passed around during creation and opening between ActivationsWriter, ActivationsWiterManager and the HDF5ActivationsWriter, but the dtype needs to be passed by the user only once during creation